src/chain/chain.rs

```diff
@@ -996,29 +996,44 @@ impl Chain {
         let is_connected = |target_name: &XorName| connected_peers.contains(&target_name);
 
         let candidates = |target_name: &XorName| {
-            let mut filtered_sections =
+            let filtered_sections =
                 self.closest_sections(target_name)
                     .into_iter()
-                    .map(|(_, members)| {
+                    .map(|(prefix, members)| {
                         (
+                            prefix,
                             members.len(),
                             members.into_iter().filter(is_connected).collect::<Vec<_>>(),
                         )
                     });
-            let best_section = filtered_sections
-                .find(|(len, connected)| connected.len() * 3 >= *len)
-                .ok_or(Error::CannotRoute);
-            best_section.map(|(len, connected)| {
-                (
-                    len,
-                    connected
-                        .into_iter()
-                        .sorted_by(|lhs, rhs| target_name.cmp_distance(lhs, rhs)),
-                )
-            })
+
+            let mut dg_size = 0;
+            let mut node_to_send = Vec::new();
```

Comment:
Should we call it `nodes_to_send`? This is a list, not a single node, after all.